qcelemental/models/molecule.py

```diff
@@ -0,0 +1,813 @@
+
+        geom_blocks = []
+        symbols = []
+        masses = []
+        real_atoms = []
+        fragments = []
+        fragment_charges = []
+        fragment_multiplicities = []
+
+        # Loop through the real blocks
+        frag_start = 0
+        for frag in real:
+            frag_size = len(self.fragments[frag])
+            geom_blocks.append(self.geometry[self.fragments[frag]])
+
+            for idx in self.fragments[frag]:
+                symbols.append(self.symbols[idx])
+                real_atoms.append(True)
+                if self._custom_masses:
+                    masses.append(self.masses[idx])
+
+            fragments.append(list(range(frag_start, frag_start + frag_size)))
+            frag_start += frag_size
+
+            fragment_charges.append(float(self.fragment_charges[frag]))
+            fragment_multiplicities.append(self.fragment_multiplicities[frag])
+
+        # Set charge and multiplicity
+        constructor_dict["molecular_charge"] = sum(fragment_charges)
+        constructor_dict["molecular_multiplicity"] = sum(x - 1 for x in fragment_multiplicities) + 1
+
+        # Loop through the ghost blocks
+        for frag in ghost:
+            frag_size = len(self.fragments[frag])
+            geom_blocks.append(self.geometry[self.fragments[frag]])
+
+            for idx in self.fragments[frag]:
+                symbols.append(self.symbols[idx])
+                real_atoms.append(False)
+                masses.append(self.masses[idx])
+
+            fragments.append(list(range(frag_start, frag_start + frag_size)))
+            frag_start += frag_size
+
+            fragment_charges.append(self.fragment_charges[frag])
+            fragment_multiplicities.append(self.fragment_multiplicities[frag])
+
+        constructor_dict["fragments"] = fragments
+        constructor_dict["symbols"] = symbols
+        constructor_dict["geometry"] = np.vstack(geom_blocks)
+        constructor_dict["real"] = real_atoms
+        constructor_dict["masses"] = masses
+
+        if orient:
+            constructor_dict["fix_orientation"] = True
+
+        return Molecule(**constructor_dict)
+
+    def to_string(self, dtype="psi4"):
+        """Returns a string that can be used by a variety of programs.
+
+        Unclear if this will be removed or renamed to "to_psi4_string" in the future
+        """
+        if dtype == "psi4":
+            return self._to_psi4_string()
+        else:
+            raise KeyError("Molecule:to_string: dtype of '{}' not recognized.".format(dtype))
+
+    def get_hash(self):
+        """
+        Returns the hash of the molecule.
+        """
+
+        m = hashlib.sha1()
+        concat = ""
+
+        tmp_json = self.to_json()
+        for field in self.hash_fields:
+            if field not in tmp_json:
+                continue
+            concat += json.dumps(tmp_json[field])
```

Comment:
As a note this will not work without the correct preparation of all of the data (rounding, zero flipping, etc).